File: pinyougou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/ItemCatService.java
package com.pinyougou.sellergoods.service;

import java.util.List;

import com.pinyougou.pojo.TbItemCat;
import entity.PageResult;
import entity.Result;

/**
 * itemCat服务层接口
 *
 * @author jieweili
 */
public interface ItemCatService {

    /**
     * 返回全部列表(已弃置)
     *
     * @return
     */
    List<TbItemCat> findAll();

    /**
     * 返回分页对象(过时)
     *
     * @param pageNum
     * @param pageSize
     * @return
     */

    PageResult findPage(Integer pageNum, Integer pageSize);

    /**
     * 添加数据
     *
     * @param itemCat
     * @return
     */
    public Result saveItemCat(TbItemCat itemCat);

    /**
     * 根据id查询一个对象
     *
     * @param id
     * @return
     */
    TbItemCat getItemCatById(Long id);

    /**
     * 修改对象
     *
     * @param itemCat
     * @return
     */
    Result updateItemCat(TbItemCat itemCat);

    /**
     * 批量删除
     *
     * @param idList
     * @return
     */
    Result removeItemCat(Long[] idList);

    /**
     * 根据条件分页查询
     *
     * @param itemCat
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult listItemCatByExample(TbItemCat itemCat, Integer pageNum, Integer pageSize);

    /**
     * 根据父id查询列表
     *
     * @param parentId
     * @return
     */
    List<TbItemCat> listItemCatByParentId(Long parentId);

}

File: pinyougou_parent/pinyougou_shop_web/src/main/webapp/js/service/itemCatService.js
app.service('itemCatService', function ($http) {
    //旧的查询方法(弃置)								   
    this.findPage = function (pageNum, pageSize) {
        return $http.get('../itemCat/findPage.do?pageNum=' + pageNum + '&pageSize=' + pageSize)
    };

    //查询方法,读取列表数据绑定到表单中
    this.search = function (pageNum, pageSize, searchEntity) {
        return $http.post('../itemCat/findPage.do?pageNum=' + pageNum + "&pageSize=" + pageSize, searchEntity)
    };

    //删除方法
    this.removeItemCatList = function (idList) {
        return $http.get('../itemCat/removeItemCat.do?idList=' + idList)
    };

    //根据id查询
    this.findOne = function (id) {
        return $http.get('../itemCat/findOne.do?id=' + id)
    };

    //修改品牌
    this.updateItemCat = function (itemCat) {
        return $http.post('../itemCat/updateItemCat.do', itemCat)
    };

    //新增品牌
    this.saveItemCat = function (itemCat) {
        return $http.post('../itemCat/saveItemCat.do', itemCat)
    };

    //根据上级id列表展示
    this.listItemCat = function (parentId) {
        return $http.get('../itemCat/listItemCat.do?parentId='+parentId);
    }
});
